src/QMCHamiltonians/OperatorBase.h

```diff
@@ -62,30 +63,31 @@ struct NonLocalData : public QMCTraits
  * Return_t is defined as RealTye.
  * The types should be checked when using complex wave functions.
  */
-struct OperatorBase : public QMCTraits
+class OperatorBase : public QMCTraits
```

Comment:
```c++
template<typename Real = QMCTraits::RealType, typename FullPrecReal = QMCTraits::FullPrecRealType>
class OperatorBase
{
...
}

Reply:
A couple of things as this might be more involved from how Operators current interact: 
1. The long term goal is to unify Real and Complex builds into a single executable (moving from compile time to run time). 
2. Because of 1. we still need a proper runtime base as template classes of different type can't be put in a single container  as currently done in the code (well technically it's possible using `void*` or `std::any`, but that's probable not desirable).